### PR TITLE
sparse cat cross entropy

### PR DESCRIPTION
**Pull request by g1y5x3**
The flat training loss was due to missing the conversion to numpy in `accuracy = (cat == y).mean()`. Saw tests passed so didn't double check. Sorry.
```
loss 2.67 accuracy 0.03: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  8.38it/s]
loss 4.37 accuracy 0.22: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 3/3 [00:00<00:00, 96.55it/s]
loss 0.09 accuracy 0.98: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 100/100 [00:01<00:00, 84.68it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 79/79 [00:00<00:00, 399.35it/s]
test set accuracy is 0.958800
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  9.20it/s]
loss 0.11 accuracy 0.98: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 200/200 [00:04<00:00, 49.17it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 79/79 [00:00<00:00, 259.02it/s]
test set accuracy is 0.971700
loss 0.23 accuracy 0.93: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 600/600 [00:03<00:00, 191.01it/s]
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 79/79 [00:00<00:00, 1138.75it/s]
test set accuracy is 0.957000
loss 2.49 accuracy 0.17: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 177.65it/s]
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 6/6 [00:00<00:00, 226.81it/s]
loss 2.21 accuracy 0.25: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 3/3 [00:00<00:00, 191.86it/s]
.
----------------------------------------------------------------------
Ran 9 tests in 9.283s
```

### CODE REVIEW

**Comment by tinyb0t** [comment]
Changes made in `tinygrad/`:
```
------------------------------------------------------------
files                             insertions       deletions
------------------------------------------------------------
tinygrad/tensor.py                         6               0
------------------------------------------------------------
lines added in the tinygrad folder: 6
```

File: extra/training.py [hunk 1]
```diff
@@ -34,7 +24,7 @@ def train(model, X_train, Y_train, optim, steps, BS=128, lossfn=sparse_categoric
     # printing
     if not noloss:
       cat = np.argmax(out.numpy(), axis=-1)
-      accuracy = (cat == y).mean()
+      accuracy = (cat == y.numpy()).mean()
```
Comment (geohot) [inline extra/training.py:27]: Hmm, how come this didn't throw an error?

Comment (g1y5x3) [reply extra/training.py:27]: It looks like the problem is on the numpy end, even if I put an Assert inside tensor.py `__eq__`, it won't able to catch it. I can putting an assert in this file, but it seems a bit pointless though. 
```
x = np.array([1,1,1])
x_tiny = Tensor([1,2,3])
print(x == x_tiny)
print((x_tiny == x).numpy())
```
[ True  True  True]
[1. 0. 0.]

Comment (ZenginU) [inline extra/training.py:27]: For this part, I have PR that uses cat.argmax() with Tensor.argmax. Maybe you dont need to convert y to numpy.

cat = np.argmax(out.numpy(), axis=-1)
accuracy = (cat == y.numpy()).mean()

if cat is not needed to be numpy than you can do.

cat = out.argmax(-1)
accuracy = (cat == y).mean().numpy()

Comment (geohot) [reply extra/training.py:27]: Oh yea, it should absolutely be this. I'll merge this, can fix in the argmax PR
